File: question_bank/two-sum/two-sum.py
# -*- coding: utf-8 -*-
# @Author  : LG

"""
执行用时：56 ms, 在所有 Python3 提交中击败了96.65% 的用户
内存消耗：15.2 MB, 在所有 Python3 提交中击败了5.48% 的用户

解题思路：
    使用字典存储，快速查找
"""
class Solution:
    def twoSum(self, nums, target: int):
        n = len(nums)
        record = {}
        for i, v in enumerate(nums):
            a = target-v
            if a not in record:
                record[v] = i
            else:
                return [i, record[a]]

File: question_bank/two-sum/two-sum.cpp
# include "vector"
#include "iostream"
#include "unordered_map"

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        for(int i=0;i<nums.size();i++){
            auto it = hashmap.find(target-nums[i]);
            if(it != hashmap.end()){
                return {it->second, i};
            }
            hashmap[nums[i]]=i;
        }
        return {};
    }
};
